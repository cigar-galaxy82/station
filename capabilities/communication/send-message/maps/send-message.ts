/* eslint-disable jest/no-export */

import { SuperfaceTest } from '@superfaceai/testing-lib';

export const sendMessage = (
  provider: string,
  params: { destination: string }
): void => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest({
      profile: 'communication/send-message',
      provider,
    });
  });

  describe(`communication/send-message/${provider}`, () => {
    it('performs correctly', async () => {
      await expect(
        superface.run({
          useCase: 'SendMessage',
          input: {
            destination: params.destination,
            text: 'Station communication/send-message test',
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
};
