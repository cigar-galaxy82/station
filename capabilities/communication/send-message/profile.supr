"""
Send Message
Send message via IM such as Messanger, Slack or MS Teams.
"""

name = "communication/send-message"
version = "2.0.0"

"""
Send Message
Sends message to one destination
"""
usecase SendMessage {
  input {
    """
    Destination
    Representation of a messaging destination. Can be ID or channel name, depending on the provider capability.
    """
    destination! string!

    """
    Text
    Text of the message
    """
    text! string!
  }

  result {
    """"
    Destination
    Identifier of the destination where the message was sent.
    """
    destination

    """
    Message ID
    Identifier of the message that was sent
    """
    messageId
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title!

    """
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail
  }
}
