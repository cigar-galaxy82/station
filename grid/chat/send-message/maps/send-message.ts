/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const sendMessageTest = (
  provider: string,
  destination: string[],
  options?: RecordingProcessOptions
): void => {
  describe(`chat/send-message/${provider}`, () => {
    let superface: SuperfaceTest;

    describe('SendMessage', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'chat/send-message',
          provider,
          useCase: 'SendMessage',
        });
      });

      describe('when specified destination does exist', () => {
        it('performs correctly', async () => {
          await expect(
            superface.run(
              {
                input: {
                  destination: destination[0],
                  text: 'test',
                  // TODO: implement sending attachments
                  attachments: [
                    {
                      id: '1',
                      createdAt: 1503435956.000247,
                      fileName: 'myfile.suma',
                      mediaType: 'text',
                      preview: 'sdasdasd',
                    },
                  ],
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });

      describe('when specified destination does not exist', () => {
        it('returns error', async () => {
          await expect(
            superface.run(
              {
                input: {
                  destination: destination[1],
                  text: 'test',
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
