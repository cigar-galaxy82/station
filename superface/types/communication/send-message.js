"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.communicationSendMessage = void 0;
var one_sdk_1 = require("@superfaceai/one-sdk");
var profile = {
    /**
     * Send Message
     * Sends message to one destination
     **/
    "SendMessage": one_sdk_1.typeHelper()
};
exports.communicationSendMessage = {
    "communication/send-message": profile
};
