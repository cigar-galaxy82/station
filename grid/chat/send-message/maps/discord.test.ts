import { sendMessageTest } from './send-message';

sendMessageTest('discord', ['935962220104396885', '000000000000000000']);
