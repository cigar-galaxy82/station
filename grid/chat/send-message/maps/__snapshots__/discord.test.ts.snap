// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/send-message/discord SendMessage when specified destination does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "destination": "935962220104396885",
    "messageId": "939113926841495623",
    "rateLimit": Object {
      "bucket": "a06de0de4a08126315431cc0c55ee3dc",
      "remainingRequests": 4,
      "remainingRequestsPercentage": 80,
      "resetTimestamp": 1643972614,
      "retryAfter": 5,
      "totalRequests": 5,
    },
  },
}
`;

exports[`chat/send-message/discord SendMessage when specified destination does not exist returns error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Not Found\\",
  \\"detail\\": \\"Unknown Channel\\"
}
AST Path: definitions[0].statements[1].responseHandlers[4].statements[0]
Original Map Location: Line 52, column 4",
}
`;
